### PR TITLE
fix(android): Handle IllegalArgumentException when Webview starts to …

### PR DESCRIPTION
**Pull request by Aldeee**
…download something but http/https URIs

# Summary

One of the libraries which we use in production make manipulation like following:

new Worker(URL.createObjectURL(new Blob(['function() {}']))).

This manipulation leads webview to make attempt to download script from the following uri: `blob:https://app.example.com/16c473d5-ac10-4b79-aa1b-4cee436e0379`. This attempt makes webview to crash with exception:

`IllegalArgumentExceptionandroid.app.DownloadManager$Request Can only download HTTP/HTTPS URIs: blob:https://app.example.com/16c473d5-ac10-4b79-aa1b-4cee436e0379`.

Might fix the following issue #295.

### What's required for testing (prerequisites)?

You can reproduce the similar behavior by implementing following code

```
import { View } from 'react-native'
import { WebView } from 'react-native-webview'

const App = () => {
  return (
    <View style={{ flex: 1 }}>
      <WebView
        source={{
          uri: 'https://codepen.io/vidhill/pen/bNPEmX'
        }}
        javaScriptEnabled
        allowFileAccess
        mixedContentMode={'always'}
        domStorageEnabled
        textZoom={150}
      />
    </View>
  )
}

export default App;
```

### What are the steps to reproduce (after prerequisites)?

Implement app with the code specified. Click "Download JSON" button inside the webview. App crashes.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 1]
```diff
@@ -196,36 +197,40 @@ protected WebView createViewInstance(ThemedReactContext reactContext) {
...
-        module.setDownloadRequest(request);
+          module.setDownloadRequest(request);
 
-        if (module.grantFileDownloaderPermissions()) {
-          module.downloadFile();
+          if (module.grantFileDownloaderPermissions()) {
+            module.downloadFile();
+          }
+        } catch (IllegalArgumentException e) {
+          System.out.println("IllegalArgumentException: " + e.toString());
```
Comment (trcoffman) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: This code fits in with the error-handling philosophy found in the rest of the method, which is to just ignore errors quietly and just log them. In that sense, this change seems fine.

Outside the scope of this PR: It seems to me like there should be some kind of callback added like `onDownloadFailedToStart`, which would alert the client of this library that a file download has failed to start.

Comment (Aldeee) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: @trcoffman thanks for you comment. Good point is to add `onDownloadFailedToStart` callback. But I've fixed concrete case here which affected us with crashes in production. Callback implementation should be done separately I suppose.

**Comment by Biplab2014** [comment]
When will it be released?

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java (hunk 1 above)
Comment (Biplab2014) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: Are you going to release it soon?

Comment (Aldeee) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: We have already publish it in our own fork https://www.npmjs.com/package/@pandadoc/react-native-webview.

**Comment by cristianoccazinsp** [comment]
Looking forward this fix as well. Does this fix any blob:// URL download attempt?

**Comment by Aldeee** [comment]
> Looking forward this fix as well. Does this fix any blob:// URL download attempt?

This fix just filters 'blob://' url from being crashed while attempting to download. It doesn't let you download blob.

**Comment by cristianocca** [comment]
Please re-open and merge. This is still an awful issue on Android.

**Pull request closed without merging**